spark/src/main/scala/geotrellis/spark/io/cog/COGValueReader.scala

```diff
@@ -71,13 +72,69 @@ trait COGValueReader[ID] {
       } catch {
         case th: Throwable => exceptionHandler(key)(th)
       }
+
+    }
+
+    def readSubsetBands(key: K, bands: Seq[Int]): Array[Option[Tile]] = {
+      val (zoomRange, spatialKey, overviewIndex, gridBounds) =
+        cogLayerMetadata.getReadDefinition(key.getComponent[SpatialKey], layerId.zoom)
+
+      val baseKeyIndex = keyIndexes(zoomRange)
+
+      val maxWidth = Index.digits(baseKeyIndex.toIndex(baseKeyIndex.keyBounds.maxKey))
+      val uri = fullPath(keyPath(key.setComponent(spatialKey), maxWidth, baseKeyIndex, zoomRange))
+
+      val sourceGeoTiff =
+        GeoTiffReader.readMultiband(uri, streaming = true)
+
+      // Have to convert the sourceGeoTiff to a GeoTiffMultibandTile
+      // in order to access the desired crop method.
+      val sourceMultibandTile =
+        GeoTiffMultibandTile(
+          sourceGeoTiff.getOverview(overviewIndex).tile,
+          sourceGeoTiff.options
+        )
+
+      // We first must determine which bands are valid and which are not
+      // before doing the crop in order to avoid band subsetting errors
+      // and/or loading unneeded data.
+      val targetBandsWithIndex: Array[(Int, Int)] =
+        bands
+          .zipWithIndex
+          .filter { case (band, _) =>
+            band >= 0 && band < sourceMultibandTile.bandCount
+          }
+          .toArray
+
+      val targetBands: Array[Int] = targetBandsWithIndex.map { _._1 }
+      val targetBandsIndexes: Array[Int] = targetBandsWithIndex.map { _._2 }
```

Comment:
You can make it in one command: 

```scala
val (targetBands, targetBandsIndexes) = targetBandsWithIndex.unzip
```

Reply:
Oh, I didn't know about that method. That does make things easier to read